agent/src/beerocks/slave/backhaul_manager/backhaul_manager_thread.cpp

```diff
@@ -2080,13 +2080,17 @@ bool backhaul_manager::handle_1905_1_message(ieee1905_1::CmduMessageRx &cmdu_rx,
     }
 }
 
-bool backhaul_manager::get_sta_bssid(const sMacAddr &sta_mac, sMacAddr &bssid)
+bool backhaul_manager::get_sta_bssid_and_socket(const sMacAddr &sta_mac, sMacAddr &bssid,
+                                                Socket **socket)
```

Comment:
It better to use a reference to a pointer (https://stackoverflow.com/a/3834103)
```suggestion
                                                Socket *&socket)
```

But why do you need to return both the `bssid` and the slave socket?
You can return the `sRadioInfo` to which the STA is connected and then extract both the `bssid` and the slave socket from there.